cmdfactory/builder_test.go

```diff
@@ -12,6 +12,99 @@ import (
+		// by bind(), which role is to assign parsed flag values to struct fields.
+		if err := cmd.Execute(); err != nil {
+			t.Fatal("Failed to execute command:", err)
+		}
+	}
+
+	type TestObj struct {
+		String string            `long:"string" usage:"String arg"`
+		Int    int               `long:"int" usage:"Integer arg"`
+		Bool   bool              `long:"bool" usage:"Boolean arg"`
+		Slice  []string          `long:"slice" usage:"Slice arg"`
+		Map    map[string]string `long:"map" usage:"Map arg"`
+		Nested struct {
+			String string            `long:"n-string" usage:"Nested string arg"`
+			Int    int               `long:"n-int" usage:"Nested integer arg"`
+			Bool   bool              `long:"n-bool" usage:"Nested boolean arg"`
+			Slice  []string          `long:"n-slice" usage:"Nested slice arg"`
+			Map    map[string]string `long:"n-map" usage:"Nested map arg"`
+		}
+	}
+
+	t.Run("String fields", func(t *testing.T) {
+		obj := &TestObj{}
+		attributeFlags(t, obj, "--string=val", "--n-string=n-val")
+		if expect, got := "val", obj.String; expect != got {
+			t.Errorf("Unexpected value for string struct field after flags attribution. Expected %q, got %q", expect, got)
+		}
+		if expect, got := "n-val", obj.Nested.String; expect != got {
+			t.Errorf("Unexpected value for nested string struct field after flags attribution. Expected %q, got %q", expect, got)
+		}
+	})
+
+	t.Run("Integer fields", func(t *testing.T) {
+		obj := &TestObj{}
+		attributeFlags(t, obj, "--int=1", "--n-int=2")
+		if expect, got := 1, obj.Int; expect != got {
+			t.Errorf("Unexpected value for int struct field after flags attribution. Expected %d, got %d", expect, got)
+		}
+		if expect, got := 2, obj.Nested.Int; expect != got {
+			t.Errorf("Unexpected value for nested int struct field after flags attribution. Expected %d, got %d", expect, got)
+		}
+	})
+
+	t.Run("Boolean fields", func(t *testing.T) {
+		obj := &TestObj{}
+		attributeFlags(t, obj, "--bool=true", "--n-bool=true")
+		if expect, got := true, obj.Bool; expect != got {
+			t.Errorf("Unexpected value for bool struct field after flags attribution. Expected %t, got %t", expect, got)
+		}
+		if expect, got := true, obj.Nested.Bool; expect != got {
+			t.Errorf("Unexpected value for nested bool struct field after flags attribution. Expected %t, got %t", expect, got)
+		}
+	})
+
+	t.Run("Slice fields", func(t *testing.T) {
+		obj := &TestObj{}
+		attributeFlags(t, obj, "--slice=val1", "--slice=val2", "--n-slice=val1,val2")
+		if expect, got := []string{"val1", "val2"}, obj.Slice; !equalSlices(got, expect) {
+			t.Errorf("Unexpected value for slice struct field after flags attribution. Expected %v, got %v", expect, got)
+		}
+		if expect, got := []string{"val1", "val2"}, obj.Nested.Slice; !equalSlices(got, expect) {
+			t.Errorf("Unexpected value for nested slice struct field after flags attribution. Expected %v, got %v", expect, got)
+		}
+	})
```

Comment:
This test is still failing.

Because flags are being parsed twice (once in cmdfactory, once in cobra), we are getting duplicated values in string slices.